pool.go

```diff
@@ -0,0 +1,273 @@
+package connection
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+type ConnectionFactoryFunc func(addr string) (*Connection, error)
+
+type Pool struct {
+	Factory ConnectionFactoryFunc
+	Addrs   []string
+	Opts    PoolOptions
+
+	mu          sync.Mutex
```

Comment:
This same mutex is protecting isClosed, connections, and full methods, need some clearer documentation listing its targets please.

Reply:
kind of fixed it :)